src/bridge/cockpitpipechannel.c

```diff
@@ -384,6 +407,10 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
       else
         self->pipe = cockpit_pipe_spawn ((const gchar **)argv, (const gchar **)env, dir, flags);
     }
+  else if (internal && lookup_internal_fd (internal, &fd))
```

Comment:
I didn't bother because you wrote this

> Yes, in theory multiple channels can be opened with the same unique string. This has similar properties to file descriptors used in process.

in bug #4380. Changed it to remove the channel and added a test.